src/llm_vm/onsite_llm.py

```diff
@@ -303,6 +304,36 @@ def model_loader(self):
         return LlamaForCausalLM.from_pretrained(self.model_uri)
     def tokenizer_loader(self):
         return LlamaTokenizer.from_pretrained(self.model_uri)
+    
+@RegisterModelClass("quantized-llama")
+class Quantized_Llama(Base_Onsite_LLM):
+    """
+    Class for running quantized Llama instances that use GGML
+
+    Attributes:
+        model_uri (str): Hugging Face Endpoint for LLM
+        model (LLM): The large language model. CTransformers includes its own tokenizer in the model
+
+    Methods:
+        model_loader: Loads the LLM into memory
+        tokenizer_loader: Does nothing. CTransformers includes its own tokenizer so this is unnecessary
+        generate: Generates a response from a given prompt with the loaded LLM and tokenizer
+    """
+    model_uri="TheBloke/LLaMa-7B-GGML"
+
+    def model_loader(self):
+        # This file specifically is the smallest model.
+        return AutoModelForCausalLM.from_pretrained(self.model_uri, model_file="llama-7b.ggmlv3.q2_K.bin")
+
+    # CTransformers loads its tokenizer in the model, so this function is unnecessary
+    def tokenizer_loader(self):
+        return
```

Comment:
no, we still need to initialize the tokenizing/detokenizing interface right? or does the stubs get set correctly?

Reply:
I _think_ ctransformers abstracts the Tokenizer away. I put up an issue asking more details on their github.

The generate function handles its own tokenizing/detokenizing and it seems to work properly, though I'm not sure if that Tokenizer is used elsewhere (I couldn't find anywhere that it is).